doc/rtd/topics/base_config_reference.rst

```diff
@@ -0,0 +1,354 @@
+.. _base_config_reference:
+
+Base Configuration
```

Comment:
Bikeshed. There are a number of places where this is defined as "system config" instead of "base config", in docs, CLI output and code.

It may be worh retaining that System Configuration name, but I can see the argument for a more clear "base config" name. 

Here is a set of places where we minimally should cross-link to this new base_configuration page if we keep "base config" naming. If we keep "system config" then I still think we need these cross-links for reference regardless of whether we start to call this Base Configuration instead of System configuration:
- doc/rtd/topics/datasources/exoscale.rst "system config"
- doc/rtd/topics/datasources/smartos.rst "system config"
- doc/rtd/topics/instancedata.rst (system_config)
- doc/rtd/topics/instancedata.rst (system config & system_config)
- doc/rtd/topics/network-config-format-v1.rst and v2

Additional code references to system config we may consider if renaming
- cloudinit/sources/__init__.py NetworkConfigSource.SYSTEM_CFG
- cloudinit/stages.py  system_config = self._extract_cfg("system")

Reply:
I think there's still a (very fuzzy) distinction between what I'm calling the base config, and the system config. To me, the base config is what we start with [at early boot](https://cloudinit.readthedocs.io/en/latest/topics/configuration.html#base-configuration), and then once everything is layered on, that forms the full system configuration. I think it's even fuzzier in the code because we may have things referencing system config that start with the base config, but also point to the final system configuration later in cloud-init's lifecycle.

I think I'd like a clearer demarcation in the docs, referring to anything in /etc/cloud as base config, but leaving that distinction fairly fuzzy in the code as long as there's no strict bounderies where one thing transforms into the other.

Does that make sense? It makes sense in my head but I'm not sure I'm describing it very well. I agree with the changes you suggested though and applied them. I would leave the code as-is, especially for the 2nd example.

Reply:
@TheRealFalcon that makes since, it's defensible and unclear. I think your PR makes great steps toward that clarification.